File: BitPositions/bp.py
# Bit Positions solution in Python 2 for CodeEval.com by Steven A. Dunn

import sys

f = open(sys.argv[1], 'r')

for line in f:
  n, pos1, pos2 = line.split(',')
  b = bin(int(n))
  if b[-int(pos1)] == b[-int(pos2)]:
    print "true"
  else:
    print "false"
f.close()

File: BitPositions/JS/bp.js
// Bit Positions solution in Node for CodeEval.com by Steven A. Dunn

var fs = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
  if (line !== "") {
    line = line.split(",");
    var n = parseInt(line[0]).toString(2);
    var pos1 = parseInt(line[1]);
    var pos2 = parseInt(line[2]);

    if (n.slice(-pos1, -pos1 + 1) == n.slice(-pos2, -pos2 + 1))
      console.log("true");
    else
      console.log("false");
  }
});
